File: pubnub-gson/src/main/java/com/pubnub/api/models/consumer/access_manager/v3/ChannelGroupGrant.java
package com.pubnub.api.models.consumer.access_manager.v3;

public class ChannelGroupGrant extends PNResource<ChannelGroupGrant> {

    private ChannelGroupGrant() {
    }

    public static ChannelGroupGrant id(String groupName) {
        ChannelGroupGrant channelGroupGrant = new ChannelGroupGrant();
        channelGroupGrant.resourceName = groupName;
        return channelGroupGrant;
    }

    public static ChannelGroupGrant pattern(String groupPattern) {
        ChannelGroupGrant channelGroupGrant = new ChannelGroupGrant();
        channelGroupGrant.resourcePattern = groupPattern;
        return channelGroupGrant;
    }

    @Override
    public ChannelGroupGrant read() {
        return super.read();
    }

    @Override
    public ChannelGroupGrant manage() {
        return super.manage();
    }
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/models/consumer/access_manager/v3/ChannelGroupGrant.kt
package com.pubnub.api.models.consumer.access_manager.v3

interface ChannelGroupGrant : PNGrant {
    companion object {
        fun id(
            id: String,
            read: Boolean = false,
            manage: Boolean = false
        ): ChannelGroupGrant = PNChannelGroupResourceGrant(
            id = id,
            read = read,
            manage = manage
        )

        fun pattern(
            pattern: String,
            read: Boolean = false,
            manage: Boolean = false
        ): ChannelGroupGrant = PNChannelGroupPatternGrant(
            id = pattern,
            read = read,
            manage = manage
        )
    }
}
